qiskit/primitives/backend_estimator_v2.py

```diff
@@ -160,9 +160,15 @@ def _validate_pubs(self, pubs: list[EstimatorPub]):
 
     def _run(self, pubs: list[EstimatorPub]) -> PrimitiveResult[PubResult]:
         pub_dict = defaultdict(list)
+        for pub in pubs:
+            data = self._preprocess_pub(pub)
```

Comment:
Please look through the `_preprocess_pub` function. It doesn't do anything relevant to your changes, so it seems like a waste to call it at all.

Reply:
I had to use _preprocess_pub to get the right data type for using .values().